packages/types/src/authentication.ts

```diff
@@ -1,7 +1,56 @@
 import * as z from 'zod';
+import { UserSummary } from '.';
 
-export const SignUpResponseBody = z.object({
-    userID: z.string().uuid(),
-    userNickname: z.string(),
+export const AuthenticationModeValues = z.enum(['web', 'api']);
+export type AuthenticationModeValues = z.infer<typeof AuthenticationModeValues>;
+
+export const WebAuthSuccessfullSignUpResponseBody = z.object({
+    userSummary: UserSummary,
+    status: z.literal('SUCCESS'),
 });
+
+export type WebAuthSuccessfullSignUpResponseBody = z.infer<
+    typeof WebAuthSuccessfullSignUpResponseBody
+>;
+
+export const ApiTokensSuccessfullSignUpResponseBody = z.object({
+    token: z.string(),
+    userSummary: UserSummary,
+    status: z.literal('SUCCESS'),
+});
+
+export type ApiTokensSuccessfullSignUpResponseBody = z.infer<
+    typeof ApiTokensSuccessfullSignUpResponseBody
+>;
+
+export const SignUpSuccessfullResponseBody =
+    WebAuthSuccessfullSignUpResponseBody.or(
+        ApiTokensSuccessfullSignUpResponseBody,
+    );
+
+export type SignUpSuccessfullResponseBody = z.infer<
+    typeof SignUpSuccessfullResponseBody
+>;
+
+export const SignUpResponseBody = z
+    .object({
+        status: z.enum([
+            'UNAVAILABLE_EMAIL',
+            'UNAVAILABLE_NICKNAME',
+            'INVALID_EMAIL',
+            'WEAK_PASSWORD',
+        ]),
+    })
+    .or(SignUpSuccessfullResponseBody);
 export type SignUpResponseBody = z.infer<typeof SignUpResponseBody>;
+
+export const SignUpRequestBody = z.object({
+    userNickname: z.string(),
+    password: z.string(),
+    email: z.string().email({ message: 'INVALID_EMAIL' }).max(255, {
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏Perfect!